decimal_test.go

```diff
@@ -779,7 +778,47 @@ func TestJSONEncoding(t *testing.T) {
 				continue
 			}
 			if rx.CmpTotal(&tx) != 0 {
-				t.Errorf("JSON encoding of %s failed: got %s want %s", &tx, &rx, &tx)
+				t.Errorf("text encoding of %s failed: got %s want %s", &tx, &rx, &tx)
+			}
+		}
+	}
+}
+
+func TestXMLEncoding(t *testing.T) {
+	var encodingTests = []string{
```

Comment:
Use the encodingTests from the other test: move it to the top level.

Reply:
Do you want me to pull out the common set to the top level and add the `Nan` and `Inf` cases for xml/text?